src/productcatalogservice/server.go

```diff
@@ -148,6 +168,12 @@ type productCatalog struct{}
 func parseCatalog() []*pb.Product {
 	var cat pb.ListProductsResponse
 
+	if problem {
+		t := time.Now().Add(150 * time.Microsecond)
+		for time.Now().Before(t) {
+		}
```

Comment:
@ahmetb Wait before merging this, I'm getting a feedback that this version of the bug changes how Stackdriver Profiler shows it. I need to retest that.

Reply:
ACK. I don't know what to do about the debugger issues yet. Let's hold until we figure out a way forward with or without the debugger.

Reply:
Ok, so I now better understand what's happening:

- That version of the bug doesn't really slow down the application.
- The previous version (with the `select`) actually slows down significantly the application, even with the same delay, but for a different reason: it consumes a lot more CPU, and with the load-generator running, we're running out of resources.

What do you think? I'd say the simplest is just to go back to that version:
```
	if problem {
		done := make(chan int)
		go func() {
			for {
				select {
				case <-done:
					return
				default:
				}
			}
		}()
		time.Sleep(50 * time.Microsecond)
		close(done)
	}
```

Reply:
Sounds good. But I still don’t quite understand why we are spinning another goroutine rather than blocking the current one. Feel free to restore the old code.